crates/codegen/parser_templates/src/shared/cst.rs

```diff
@@ -0,0 +1,131 @@
+use std::ops::Range;
+use std::rc::Rc;
+
+use serde::Serialize;
+
+use super::kinds::*;
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
```

Comment:
In both CST nodes and visitors, I wanted to suggest using `range` instead of `span`. This has the benefit of matching its actual `std::ops::Range` type.

Additionally, from what I have seen so far, "Span" is used to:

1. Indicate the difference between the range's start and end. In parsing, that would be the token length. The token `bar` in `foo(bar)` would have a range of `4..7` and a span of `3`, as it "spans" 3 characters.
2. In many rust crates (and even [rustc](https://github.com/rust-lang/rust/blob/f9216b75646cde0c4c69ae00232778a47fc893d3/compiler/rustc_span/src/span_encoding.rs#L76)), it is used to describe a data structure that holds additional metadata about the actual contents of a node, not just the numeric start/end indices.

Thoughts?

Reply:
ok, agreed